### PR TITLE
[subset] Various fixes for problems reported by oss-fuzz

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: src/hb-ot-glyf-table.hh [hunk 1]
```diff
@@ -232,9 +232,11 @@ struct glyf
   {
     inline void init (hb_face_t *face)
     {
+      memset (this, 0, sizeof (accelerator_t));
+
       hb_blob_t *head_blob = Sanitizer<head>().sanitize (face->reference_table (HB_OT_TAG_head));
       const head *head_table = Sanitizer<head>::lock_instance (head_blob);
-      if ((unsigned int) head_table->indexToLocFormat > 1 || head_table->glyphDataFormat != 0)
+      if (!head_table || (unsigned int) head_table->indexToLocFormat > 1 || head_table->glyphDataFormat != 0)
```
Comment (behdad) [inline src/hb-ot-glyf-table.hh]: Humm. I thought head_table will be the null table in case of failure?

**Comment by behdad** [comment]
lgtm if bots happy. Thanks!

File: src/hb-ot-glyf-table.hh (hunk 1 above)
Comment (garretrieger) [reply src/hb-ot-glyf-table.hh]: Fixed.

File: src/hb-ot-glyf-table.hh [hunk 2]
```diff
@@ -236,7 +236,7 @@ struct glyf
 
       hb_blob_t *head_blob = Sanitizer<head>().sanitize (face->reference_table (HB_OT_TAG_head));
       const head *head_table = Sanitizer<head>::lock_instance (head_blob);
-      if (!head_table || (unsigned int) head_table->indexToLocFormat > 1 || head_table->glyphDataFormat != 0)
+      if (head_table == &Null(head) || (unsigned int) head_table->indexToLocFormat > 1 || head_table->glyphDataFormat != 0)
```
Comment (behdad) [inline src/hb-ot-glyf-table.hh]: Technically we shouldn't need the first check. If head_table is Null, then indexToLocFormat will be 0. So, just change the second check to check for == 1?

Comment (behdad) [reply src/hb-ot-glyf-table.hh]: Oh, my bad. Yeah maybe this is find.

Comment (garretrieger) [reply src/hb-ot-glyf-table.hh]: Fixed.

**Pull request merged**